cpp/src/groupby/hash/groupby.cu

```diff
@@ -594,22 +591,21 @@ std::unique_ptr<table> groupby_null_templated(table_view const& keys,
                                               rmm::cuda_stream_view stream,
                                               rmm::mr::device_memory_resource* mr)
 {
-  auto d_keys = table_device_view::create(keys, stream);
-  auto map    = create_hash_map<keys_have_nulls>(*d_keys, include_null_keys, stream);
+  auto d_keys_ptr = table_device_view::create(keys, stream);
+  auto hash_map   = create_hash_map<keys_have_nulls>(*d_keys_ptr, include_null_keys, stream);
 
   // Cache of sparse results where the location of aggregate value in each
   // column is indexed by the hash map
   cudf::detail::result_cache sparse_results(requests.size());
 
   // Compute all single pass aggs first
   compute_single_pass_aggs<keys_have_nulls>(
-    keys, requests, &sparse_results, *map, include_null_keys, stream);
+    keys, requests, &sparse_results, *hash_map, include_null_keys, stream);
 
   // Extract the populated indices from the hash map and create a gather map.
   // Gathering using this map from sparse results will give dense results.
-  auto map_and_size = extract_populated_keys(*map, keys.num_rows(), stream);
-  rmm::device_uvector<size_type> gather_map{std::move(map_and_size.first)};
-  size_type const map_size = map_and_size.second;
+  auto [map, map_size] = extract_populated_keys(*hash_map, keys.num_rows(), stream);
+  rmm::device_uvector<size_type> gather_map{std::move(map)};
```

Comment:
It's a shame that the CTAD change didn't pan out. Should we remove mention of CTAD from the commit message/title?